### PR TITLE
Fix building on macOS with clang

### PR DESCRIPTION
**Pull request by Karsten1987**
as mentioned in https://github.com/ros-controls/ros2_control/pull/275#issuecomment-759131671, the use of Linux posix API doesn't work on OSX.
Why not using standard C++ for this?

this pr also addresses some warnings in the transmission_interface package.

### CODE REVIEW

**Comment by Karsten1987** [comment]
The linter GH actions are still not properly configured. They should target `foxy` as their distribution. I believe they default to rolling.

**Review by destogl — CHANGES_REQUESTED** [review]
It looks good, but why is not building?

File: transmission_interface/test/simple_transmission_test.cpp [hunk 1]
```diff
@@ -174,7 +174,8 @@ INSTANTIATE_TEST_CASE_P(
     SimpleTransmission(10.0, 1.0),
     SimpleTransmission(10.0, -1.0),
     SimpleTransmission(-10.0, 1.0),
-    SimpleTransmission(-10.0, -1.0)));
+    // cppcheck-suppress syntaxError
```
Comment (bmagyar) [inline transmission_interface/test/simple_transmission_test.cpp:177]: please remove this.

It isn't related to fixing the thread, I'd really like to not have manual linter management in code

Comment (Karsten1987) [reply transmission_interface/test/simple_transmission_test.cpp:177]: please propose an alternative fix for this then. It does not compile/pass tests without this.

**Comment by Karsten1987** [comment]
> but why is not building?

as mentioned, `clock_nanosleep`, this is using Linux Posix API which is not available on OSX.

**Comment by destogl** [comment]
> as mentioned, `clock_nanosleep`, this is using Linux Posix API which is not available on OSX.

That's OK, bit why CI has issues?

**Comment by bmagyar** [comment]
> That's OK, bit why CI has issues? 

because it automatically updated to rolling... :facepalm:

**Comment by bmagyar** [comment]
First of all, I would like to state that I am generally unhappy when we have to work around linters by adding comments to our code. It is noise that disturbs whoever tries to read said code and shouldn't be there.

With that out of the way, let's resolve this.
@Karsten1987 I've created two PRs https://github.com/ros-controls/ros2_control/pull/296 and https://github.com/ros-controls/ros2_control/pull/297 . By merging either of them, I'm "happy" to go ahead with this PR.

If you'd like to avoid "wat dis" type of questions during a review, please name your PRs more accurately. You are working with a platform the rest of us are not actively using and we don't have a CI for, hence these things are far from evident at the moment.

File: transmission_interface/test/simple_transmission_test.cpp (hunk 1 above)
Comment (bmagyar) [reply transmission_interface/test/simple_transmission_test.cpp:177]: https://github.com/ros-controls/ros2_control/pull/296 or https://github.com/ros-controls/ros2_control/pull/297, please pick

**Comment by bmagyar** [comment]
@emersonknapp do you have any suggestions for fixing the linters to a Foxy version?

**Comment by emersonknapp** [comment]
> @emersonknapp do you have any suggestions for fixing the linters to a Foxy version?

Yes, you can add `distribution: foxy` in the `with` statement (https://github.com/ros-controls/ros2_control/blob/master/.github/workflows/lint.yml#L31) - see https://github.com/ros-tooling/action-ros-lint/blob/master/action.yml#L12

File: transmission_interface/test/simple_transmission_test.cpp (hunk 1 above)
Comment (Karsten1987) [reply transmission_interface/test/simple_transmission_test.cpp:179]: done

**Comment by bmagyar** [comment]
thanks guys!

**Review by bmagyar — APPROVED** [review]
thanks for the fixes!

**Comment by destogl** [comment]
@bmagyar could you remove the merged branch? We should keep repo clean :)

**Comment by bmagyar** [comment]
indeed. we shouldn't even push branches to this repo in the first place

**Pull request merged**